packages/protoc-gen-ng/src/output/types/message.ts

```diff
@@ -69,10 +69,13 @@ export class Message {
       }
     }
 
-    this.oneOfs = this.message.oneofDeclList.map(od => new OneOf(this.proto, this.message, od));
+    const allOneOfs = this.message.oneofDeclList.map(od => new OneOf(this.proto, this.message, od));
```

Comment:
Here we have to list, since we need the original oneof, where we can access the correct oneof object for the `MessageField`. And then the filtered one where we remove the synthetic optional oneof instances, so that it wouldn't generate any of the oneof code logic.